lib/classes/PluginManager.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const Utils = require('./Utils');
+const path = require('path');
+const has = require('lodash').has;
+const forEach = require('lodash').forEach;
+
+class PluginManager {
+  constructor(serverless) {
+    this._serverless = serverless;
+    this._pluginInstances = [];
+    this._commandsList = [];
+    this._commands = {};
+  }
```

Comment:
Also, you should probably add a test for each class constructor that checks that the serverless instance get attached. Since the serverless instance is critical for the class to work. I started to do that for all of my classes as well